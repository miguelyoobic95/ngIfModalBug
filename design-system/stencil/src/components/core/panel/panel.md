---
name: Panel
category: Core Components
---
Panel are use to help separating content a making processes as reading or filling a form easier to the user. Panels are a way of chunking into smaller pieces.

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`width`|number|`100%`|the width of the panel|
|`height`|number|`100%`|the height of the panel|

### Basic Panel

```yoo-panel.html
    <yoo-panel></yoo-panel>
```

### Different Sized Panel

```yoo-panel-sizes.html
    <yoo-panel height="100"></yoo-panel>
    <br>
    <yoo-panel height="100" width="100"></yoo-panel>
```

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|yoo-panel accent|
|Dark|`dark`|yoo-panel dark|
|Success|`success`|yoo-panel success|
|Danger|`danger`|yoo-panel danger|
|Info|`info`|yoo-panel info|
|Warning|`warning`|yoo-panel warning|
|Gradient Accent|`gradient-accent`|yoo-panel gradient-accent|
|Gradient Dark|`gradient-dark`|yoo-panel gradient-grey|
|Gradient Success|`gradient-success`|yoo-panel gradient-success|
|Gradient Danger|`gradient-danger`|yoo-panel gradient-danger|
|Gradient Info|`gradient-info`|yoo-panel gradient-info|
|Gradient Warning|`gradient-warning`|yoo-panel gradient-warning|

```yoo-panel-styles.html
    <yoo-panel height="50" class="accent"></yoo-panel>
    <br>
    <yoo-panel height="50" class="danger"></yoo-panel>
    <br>
    <yoo-panel height="50"class="success"></yoo-panel>
    <br>
    <yoo-panel height="50" class="info"></yoo-panel>
    <br>
    <yoo-panel height="50" class="warning"></yoo-panel>
    <br>
    <yoo-panel height="50" class="dark"></yoo-panel>
    <br>
    <yoo-panel height="50" class="gradient-accent"></yoo-panel>
    <br>
    <yoo-panel height="50" class="gradient-danger"></yoo-panel>
    <br>
    <yoo-panel height="50" class="gradient-success"></yoo-panel>
    <br>
    <yoo-panel height="50" class="gradient-info"></yoo-panel>
    <br>
    <yoo-panel height="50" class="gradient-warning"></yoo-panel>
    <br>
    <yoo-panel height="50" class="gradient-dark"></yoo-panel>
```
