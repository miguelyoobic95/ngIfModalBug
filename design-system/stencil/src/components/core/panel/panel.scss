

@mixin color($color) {
    .outer-container {
        background: cssvar($color);
    }
}

:host {
    .outer-container {
        background: cssvar(light);
        &.width {
            width: 100%;
        }
        &.height {
            height: 100%;
        }
    }
}

:host(.accent) {
    @include color(accent);
}

:host(.danger) {
    @include color(danger);
}

:host(.success) {
    @include color(success);
}

:host(.info) {
    @include color(info);
}

:host(.warning) {
    @include color(warning);
}

:host(.dark) {
    @include color(dark);
}

:host(.gradient-accent) {
    @include color(gradient-accent);
}

:host(.gradient-danger) {
    @include color(gradient-danger);
}

:host(.gradient-success) {
    @include color(gradient-success);
}

:host(.gradient-info) {
    @include color(gradient-info);
}

:host(.gradient-warning) {
    @include color(gradient-warning);
}

:host(.gradient-dark) {
    @include color(gradient-dark);
}