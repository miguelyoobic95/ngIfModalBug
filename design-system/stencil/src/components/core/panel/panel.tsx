import { Component, Prop, Element } from '@stencil/core';

@Component({
    tag: 'yoo-panel',
    styleUrl: 'panel.scss',
    scoped: true
})
export class YooPanelComponent {

    @Prop() width: number;
    @Prop() height: number;

    @Element() host: HTMLStencilElement;

    render(): JSX.Element {
        return (
            <div>
                <canvas class={'outer-container' + (this.width ? '' : ' width' ) + (this.height ? '' : ' height')} width={this.width} height={this.height}></canvas>
            </div>
        );
    }
}
