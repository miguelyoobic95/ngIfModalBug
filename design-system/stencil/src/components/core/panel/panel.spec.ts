import { TestWindow } from '@stencil/core/testing';
import { YooPanelComponent } from './panel';

describe('YooPanelComponent', () => {
    it('should build', () => {
        expect(new YooPanelComponent()).toBeTruthy();
    });

    describe('Snapshots', () => {
        it ('Should render', async () => {
            let window = new TestWindow();
          let element = await window.load({
                components: [YooPanelComponent],
                html: '<yoo-panel></yoo-panel>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should render with width', async () => {
            let window = new TestWindow();
          let element = await window.load({
                components: [YooPanelComponent],
                html: '<yoo-panel width="100"></yoo-panel>'
            });
            expect(element).toMatchSnapshot();
        });

         it ('Should render with height', async () => {
            let window = new TestWindow();
          let element = await window.load({
                components: [YooPanelComponent],
                html: '<yoo-panel height="100"></yoo-panel>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('Should render with width and height', async () => {
            let window = new TestWindow();
          let element = await window.load({
                components: [YooPanelComponent],
                html: '<yoo-panel width="100" height="100"></yoo-panel>'
            });
            expect(element).toMatchSnapshot();
            await window.flush();
            expect(element.getElementsByClassName('outer-container')).toHaveLength(1);
        });
    });

    describe('rendering', () => {
        let element;
        beforeEach( async () => {
          let window = new TestWindow();
          element = await window.load({
            components: [YooPanelComponent],
            html: '<yoo-panel></yoo-panel>'
          });
        });
        it ('should render', async () => {
            expect(element).toBeTruthy();
        });
    });
});
