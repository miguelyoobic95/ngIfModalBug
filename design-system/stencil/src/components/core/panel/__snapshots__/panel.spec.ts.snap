// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooPanelComponent Snapshots Should render 1`] = `
<yoo-panel
  class="hydrated"
  data-yoo-panel-host=""
>
  
  <div
    data-yoo-panel=""
  >
    <canvas
      class="outer-container width height"
      data-yoo-panel=""
    />
  </div>
</yoo-panel>
`;

exports[`YooPanelComponent Snapshots Should render with height 1`] = `
<yoo-panel
  class="hydrated"
  data-yoo-panel-host=""
  height="100"
>
  
  <div
    data-yoo-panel=""
  >
    <canvas
      class="outer-container width"
      data-yoo-panel=""
      height="100"
    />
  </div>
</yoo-panel>
`;

exports[`YooPanelComponent Snapshots Should render with width 1`] = `
<yoo-panel
  class="hydrated"
  data-yoo-panel-host=""
  width="100"
>
  
  <div
    data-yoo-panel=""
  >
    <canvas
      class="outer-container height"
      data-yoo-panel=""
      width="100"
    />
  </div>
</yoo-panel>
`;

exports[`YooPanelComponent Snapshots Should render with width and height 1`] = `
<yoo-panel
  class="hydrated"
  data-yoo-panel-host=""
  height="100"
  width="100"
>
  
  <div
    data-yoo-panel=""
  >
    <canvas
      class="outer-container"
      data-yoo-panel=""
      height="100"
      width="100"
    />
  </div>
</yoo-panel>
`;
